src/libraries/System.Private.Xml/src/System/Xml/Xslt/XslCompiledTransform.cs

```diff
@@ -89,6 +89,7 @@ private void Reset()
 
         // SxS: This method does not take any resource name and does not expose any resources to the caller.
         // It's OK to suppress the SxS warning.
+        [RequiresUnreferencedCode("This method requires a call to LoadInternal which might use types that cannot be statically analyzed.")]
```

Comment:
So this means there's no way to use XSLT and be "trim compatible" right?
I haven't looked into the details - maybe that's really the case. In any case I would probably create an issue to track this -  in theory running XSLT should not break trimming (precompiled XSLT is a separate problem) - in most cases.

Reply:
Yeah, I wasn't familiar with xslt but have been playing around with the scenarios, and there is actually only one scenario that should be unsafe:
We have a tool called xsltc.exe that takes in an xsl file and compiles it into a generated managed assembly, and we have an entry point from XslCompiledTransform.Load that can take in a pointer to this generated managed assembly and will internally call methods inside that assembly using reflection, which will of course be not trim safe. The reason why I had to annotate all Load methods is because they internally all use a common LoadInternal which I had to annotate as it also has this unsafe scenario included in it, but I will refactor so that only this specific scenario is marked as unsafe but not the rest.

Reply:
I'd assume that if you want to use XslCompiledTransform you'd likely need most of the types from Xsl.Runtime namespace.

Reply:
If there are dependencies those should be either found by the trimmer automatically, or we will need to add trimmer annotations (like `DynamicDependency`). If it ends up including most of `Xsl.Runtime`, so be it. The point of this discussion is that it should be possible to run XSLT even in a trimmed app without issues. Unless it uses precompiled XSLT, that won't work.

Reply:
> that it should be possible to run XSLT even in a trimmed app without issues.

I made at least the simplest case work in #45522, and added a trimming test for it.

Reply:
Yeah, if you guys check my latest change on this PR, I've set it so that only two Load overloads are marked as RUC which are the ones that take in a Type and a MethodInfo of an assembly that was generated by xsltc.exe when passed in a xsl. This will of course always be unsafe, but all other Load overloads are fine because even though some of the types they instantiate cannot be statically analyzed, all of these types are generated using ref emit at runtime so the trimmer can't remove those. Most of Xsl.Runtime is preserved, but that is just because the linker can statically analyze that it needs it so no issues there.